Taglierino.hs

```diff
@@ -949,7 +1067,9 @@ compileWith opts@Options{..} sys =
           hPrint h $ pretty def
         hPutStrLn h "// Attacker"
         hPrint h $ pretty $ attacker opts pPublic pAllowed pKnowledgeSize
-        hPrint h $ pretty $ LTS.Parallel "System" $ LTS.Primitive $ "Attacker" : map (LTS.name . fst) (M.elems compiledAgents)
+        hPutStrLn h "// Storages"
+        forM_ (M.toList pStore) (\(name, records)-> hPrint h $ pretty $ makeStore pAllowed records name (numInstance name pProcs)) 
+        hPrint h $ pretty $ LTS.Parallel "System" $ LTS.Primitive $ ("Attacker" : map (LTS.name . fst) (M.elems compiledAgents)) ++ map ("Store_" ++) (M.keys pStore)
```

Comment:
We could split this expression using intermediate variables, so that it is easier to read.